src/components/OSMElementData/OSMElementData.jsx

```diff
@@ -0,0 +1,196 @@
+    const activeFeatureId = selectedFeatureId ? selectedFeatureId : featureIds[0]
+
+    // If we're already fetching data for the active feature, or if the fetch failed,
+    // there's nothing to do
+    if (fetchingElement === activeFeatureId || failedElement === activeFeatureId) {
+      return
+    }
+
+    else if (_isEmpty(featureIds)) {
+      // TODO: should I call setElement(null)?
+      // No features to fetch, so nothing to do
+      return
+    } else if (element) {
+      return
+    }
+
+    setFetchingElement(activeFeatureId)
+    setFailedElement(null)
+    setFetchErr(null)
+    fetchOSMElement(activeFeatureId).then(element => {
+      // If for some reason we don't get any element entries, record a failure
+      // to prevent further attempts to refetch
+      if (!element) { // TODO Can this happen ?
+        setFailedElement(activeFeatureId)
+        setFetchingElement(null)
+        return
+      }
+
+      // Sort history entries by version, reversing to get descending order
+      setElement(element) // TODO: Should I store it to avoid other calls when the user switch the feature?
+      setFetchingElement(null)
+    }).catch(err => {
+      setFailedElement(activeFeatureId)
+      setFetchErr(err)
+      setFetchingElement(null)
+    })
+  }, [selectedFeatureId, featureIds, fetchingElement, failedElement, fetchOSMElement])
+
+  if (fetchingElement) {
+    return (
+      <div className="mr-flex mr-justify-center mr-items-center mr-w-full mr-h-full">
+        <BusySpinner />
+      </div>
+    )
+  }
+
+  const activeFeatureId = selectedFeatureId ? selectedFeatureId : featureIds[0]
+  if (failedElement === activeFeatureId) {
+    return (
+      <div className="mr-flex mr-flex-col mr-text-red-light">
+        <FormattedMessage {...messages.elementFetchFailed} values={{element: activeFeatureId}} />
+        {fetchErr && fetchErr.defaultMessage && <FormattedMessage {...fetchErr} />}
+      </div>
+    )
+  }
+
+  if (!element) {
+    return (
+      <FormattedMessage {...messages.noOSMElements} /> // TODO: What to display here?
+    )
+  }
+
+  const tagsRows = _map(element.tag, tag => <tr key={tag.k}><td>{tag.k}</td><td>{tag.v}</td></tr>)
+
+  return (
+    <div className="mr-mr-4">
+      <div className="mr-flex mr-justify-between mr-links-green-lighter mr-mb-4">
+        <FeatureSelectionDropdown
+          featureIds={featureIds}
+          selectedFeatureId={activeFeatureId}
+          selectFeatureId={setSelectedFeatureId}
+        />
+        <a
+          className="mr-button mr-button--xsmall"
+          href={`${OSM_SERVER}/${activeFeatureId}`}
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          <FormattedMessage {...messages.viewOSMLabel} />
+        </a>
+      </div>
+
+      <table className="mr-w-full">
+        <thead>
+          <tr>
+            <th><FormattedMessage {...messages.headerTag} /></th>
+            <th><FormattedMessage {...messages.headerValue} /></th>
```

Comment:
For some reasons this does not work, I see a  "missing ìd` property" from the translation module.